### ISSUE TITLE
Document improvement suggestions

### ISSUE THREAD

**Issue by whinc** [mobxjs/mobx#2192]
Welcome to MobX. Please provide as much relevant information as possible!

I have a:

1. [x] Question: Feel free to just state your question. For a quick answer, there are usually people online at our [Gitter](https://gitter.im/mobxjs/mobx) channel
2. [ ] Documentation improvement. Please don't open an issue but create a PR instead!
2. [ ] Issue:
  * [ ] **Provide error messages including stacktrace**
  * [ ] Provide a **minimal** sample reproduction. **Create a reproduction based on this [sandbox](https://codesandbox.io/s/v3v0my2370)**
  * [ ] Did you check this issue wasn't filed before?
  * [ ] Elaborate on your issue. What behavior did you expect?
  * [ ] State the versions of MobX and relevant libraries. Which browser / node / ... version?
3. [ ] Idea:
  * [ ] What problem would it solve for you?
  * [ ] Do you think others will benefit from this change as well and it should in core package (see also mobx-utils)?
  * [ ] Are you willing to (attempt) a PR yourself?

Please tick the appropriate boxes. Feel free to remove the _other_ sections.

**Please be sure to close your issues promptly.**

Now that the documentation is not friendly enough for beginners, it is recommended to move the decorator syntax and V4 version related content to separate chapters to keep MobX core content streamlined.

**Comment by FredyC** [comment]
Care to elaborate? What is V4 related content? Why should be decorator moved somewhere else?

**Comment by whinc** [comment]
I have been learning MobX for a while and I have been using it for a while. I mainly read the official documents and found that some places in the learning process will increase the learning curve.

First, I found that decorators (such as @observable, @observer, @action, etc.) run through the entire document, from the API to the examples everywhere, as if the decorator is one of the core features of MobX, in fact it is not, but it occupies A lot of attention. Especially when I first came into contact with MobX, many new concepts (such as observable, deravation, etc.) need to be digested, which I think is the most important.

Second, the decorator is almost useless in the new React Hooks project. And the decorator specification is still changing, I don't consider introducing the decorator to increase the cost of understanding the code (I admit it is good).

Finally, some parts of the documentation will have some considerations about the V4 version. For example, dynamically adding an observable state requires the use of set/get provided by MobX, etc. Actually, there is no such problem in the V5+ version, but when reading the documentation, I need to Keep an eye on these tips and worry about missing some nuances between them. If I have adopted the V5+ version, I don't have to worry about it at all, but now I need it.

In general, I hope that anyone can learn the core content of MobX from the documentation, and will not be affected by non-core content (such as decorators, V4 notes in V5) during the learning process.

**Comment by FredyC** [comment]
Thanks for the feedback, always appreciated.

Personally, I don't use decorators either, it's just weird syntax to my taste and doesn't fit well into TypeScript world (the mobx-state-tree is better option imo). However, my understanding is, that decorators still do have their place in the code in case you want to declare a store outside of React components. Sure, there are nifty looking hooks like `useLocalStore`, but there is some uncertainty hanging about Concurrent mode. It's unclear if it will work correctly in future versions of React. The external stores with decorators probably don't have any issues in that field.

I agree that official https://mobx.js.org/ documentation can be confusing and perhaps outdated in some places. Sometimes even misleading. Lately, we have migrated to the Docusaurus platform so it's easier for other people to contribute here. MobX is open source and it can only improve by the work of other people - volunteers who can help out to make things better.

For someone who is using MobX for several months or even years, it's kinda hard to see these shortcomings and just go and fix it. That's why we need other people who still have fresh eyes on the problem and can contribute some improvements. We can review and help out as much as we can, but it's unlikely we will just go and rewrite whole docs on our own.

Another point of view is that there are pioneers who adopted Hooks straight away (I certainly did) and left everything else in the dust. However, there are also people coming to big corporate projects that might be still using MobX 3/decorators/classes/whatever and we cannot just delete these portions of docs because we think it's outdated. They need to be able to reach out and find what do these things do.

This is also one of the reasons why I made https://mobx-react.js.org/ where I wanted to consolidate the knowledge that is more future-proof and leaving some of that old stuff in dust. Sure, it's far from perfect, but once again, contributions from other people are what can make things better. One person cannot ever explain everything to be understandable by everyone.

---

Regarding the V4/V5 differences. We are certainly trying to keep V4 up to date as much as possible with fixes and features. It's daunting tasks by itself, but it's a necessary evil. If you have some idea for a documentation page that will summarize differences, the contribution would be much appreciated.
